### PR TITLE
Line Display Options

### LINKED ISSUE #104: Ability to customise style of line plots
In the plot options menu we neeed to allow 1D line plots to be customised. This should include:
* line colour
* line style
* line width 
* marker type

These options should only be shown if the plot is of the appropriate type (1D line plot).

### LINKED ISSUE #109: Option to toggle lines on/off for cut plots
Legend titles can currently be toggled on or off in plot options. It should be possible to also toggle lines on/off.

### PR DESCRIPTION
**Pull request by MatthewBowles**
Added various options to change the appearance of lines in cut plots.
- Refactored legends code to be more extendable and removed redundant data
- Added controls to UI for line appearance ( line colour, style, width, marker type and show/hide)
- Add unit test to presenter for new line controls

**To test:**
- Open any cut plot
- Click plot options
- Change values for line colour / style / width / marker and click OK

Fixes #104 
Fixes #109

### CODE REVIEW

File: mslice/plotting/plot_window/plot_figure.py [hunk 1]
```diff
@@ -179,11 +161,75 @@ def _toggle_errorbars(self):
...
+        handles = current_axes.get_legend_handles_labels()[0]
+        for i in range(len(legends)):
+            container = current_axes.containers[i]
+            container.set_label(legends[i]['label'])
+            if legends[i]['visible']:
+                handles_to_show.append(handles[i])
+                labels.append(legends[i]['label'])
+            self.legends_visible[i] = legends[i]['visible']
+        x = current_axes.legend(handles_to_show, labels)  # add new legends
+        x.draggable()
```
Comment (martyngigg) [inline mslice/plotting/plot_window/plot_figure.py]: Good. It's useful being able to move the legend around.

**Comment by MatthewBowles** [comment]
A scroll bar is now added to the legend options when there are three or more cuts. This was the simplest solution to implement.
Also, if a line colour is selected that is the same as an existing one, a warning message is displayed and it reverts to the previous selection.

**Comment by MatthewBowles** [comment]
@martyngigg  I moved the legends to the right hand side as discussed, added the toggling for line visibility and made the show/hide controls a bit clearer.
Let me know if there is anything else that could be improved.

**Comment by martyngigg** [comment]
@MatthewBowles Thanks.

I think this looks better but there seems to be a bit of a problem with the resize behaviour of the dialog. Does the UI file contain a layout?

mslice-plot-options-resize

The show behaviour looks nice and I've only found a minor issue:

* if I uncheck one of lines to show
* click okay
* click plot options again
* click "show legend" will just check the legend box but not cause "show" to become set - we should either keep the legend box disabled as before or enable show when show legend is clicked

**Pull request merged**